app/src/main/java/com/col/eventradar/ui/SearchFragment.kt

```diff
@@ -0,0 +1,207 @@
+package com.col.eventradar.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.col.eventradar.databinding.FragmentSearchBinding
+import com.col.eventradar.ui.adapters.SearchResultsAdapter
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.json.JSONArray
+import java.net.HttpURLConnection
+import java.net.URL
+import java.net.URLEncoder
+
+class SearchFragment : Fragment() {
+
+    private var bindingInternal: FragmentSearchBinding? = null
+    private val binding get() = bindingInternal!!
+    private var listener: MapFragmentListener? = null
+    private val handler = Handler(Looper.getMainLooper())
+    private var searchRunnable: Runnable? = null
+    private var isProgrammaticChange = false
+
+    private val searchResultsAdapter = SearchResultsAdapter { result ->
+        listener?.onLocationSelected(result)
+        isProgrammaticChange = true
+        binding.searchEditText.setText(result.name)
+        binding.searchResultsRecyclerView.visibility = View.GONE
+        binding.searchEditText.clearFocus()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        bindingInternal = FragmentSearchBinding.inflate(inflater, container, false)
+
+        binding.searchResultsRecyclerView.layoutManager = LinearLayoutManager(requireContext())
+        binding.searchResultsRecyclerView.adapter = searchResultsAdapter
+
+        binding.searchEditText.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
```

Comment:
```suggestion
            override fun afterTextChanged(searchValue: Editable?) {
```
would give it more meaningful name